### PR TITLE
Assistant Messages and Layout

### PR DESCRIPTION
**Pull request by Hamzali**
I have added `react-native-linear-gradient` for gradient background for message bubble and option buttons. If there is a better way to achieve gradient background please share.

I couldn't decide about the color scheme yet therefor, I left colors in place will update with your directions.

Used React Native's StyleSheet for native style reuse on components instead of plain objects therefore I needed to change functional styles with memoize.

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -310,15 +312,18 @@ PODS:
     - React-jsi (= 0.63.4)
   - RNCAsyncStorage (1.13.2):
     - React
+  - RNGestureHandler (1.9.0):
+    - React-Core
   - RNKeychain (6.2.0):
     - React
-  - RNSVG (12.1.0):
+  - RNSVG (9.13.6):
```
Comment (gwn) [inline ios/Podfile.lock]: Don't really care, but just curious. Any ideas why this change happened? Looks like the new `react-native-linear-gradient` dependency triggered a downgrade on this native module `RNSVG`, which is probably ok for now. Just wanted to hear from you if you have any deeper knowledge regarding this than me. But if not, no problem at all.

File: ios/ixoWallet.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>IDEDidComputeMac32BitWarning</key>
+	<true/>
+</dict>
+</plist>
```
Comment (gwn) [inline ios/ixoWallet.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:8]: Again, don't really care but just curious. Any ideas about what this file is?

File: src/assistant/Assistant.js [hunk 3]
```diff
@@ -1,11 +1,23 @@
+// const {default: AssistantLogo} = require('../assets/top-assistant-icon.svg')
+// const {default: CloseIcon} = require('../assets/close-icon.svg')
+// const {default: ScanIcon} = require('../assets/qr-scan.svg')
+
```
Comment (gwn) [inline src/assistant/Assistant.js]: Deleting is fine, no need to comment out. I guess you either forgot or left them because this patch is still a work in progress. Both are ok btw, no problem, just leaving a note.

File: src/assistant/Assistant.js [hunk 4]
```diff
@@ -1,11 +1,23 @@
...
-    handleCustomAssistantResponse = require('./assistantResponseHandler')
+    {Modal, Text, QRScanner} = require('$/lib/ui'),
+    handleCustomAssistantResponse = require('./assistantResponseHandler'),
+    theme = require('$/lib/theme'),
+    MessageBubble = require('$/assistant/MessageBubble'),
+    ScanIcon = require('$/lib/icons/Scan'),
+    CloseIcon = require('$/lib/icons/Close'),
+    AssistantIcon = require('$/lib/icons/Assistant'),
+    ArrowUpIcon = require('$/lib/icons/ArrowUp'),
+    {default: LinearGradient} = require('react-native-linear-gradient')
```
Comment (gwn) [inline src/assistant/Assistant.js]: This having to import modules with the `default` keyword from time to time is really bothering me. This is because of my decision of using the CommonJS syntax over the ES Modules syntax, just because I prefer the client codebases to look like Node codebases as much as possible. But I can't deny this problem, although I can live with it. What about you? Any thoughts?

File: src/lib/theme.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    background: '',
+    white: '',
+    black: '',
+}
+
+module.exports =  {
+    spacing,
+    shadow,
+    colors,
+}
```
Comment (gwn) [inline src/lib/theme.js:28]: Cool, exactly the way I imagined this.

File: src/lib/theme.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+const BASE_MEASURE = 8
+
+const spacing = (n) => n * BASE_MEASURE
+
+const shadow = () => ({
```
Comment (gwn) [inline src/lib/theme.js:5]: I guess you're planning to parameterize this at some point, I assume you wouldn't implement this as a function otherwise. Am I right?

**Comment by gwn** [comment]
> I have added react-native-linear-gradient for gradient background for message bubble and option buttons. If there is a better way to achieve gradient background please share.

I don't know, but this looks fine. Thanks for asking.

> I couldn't decide about the color scheme yet therefor, I left colors in place will update with your directions.

All right, I don't know how to proceed there too for now, we can make a decision together in the following days.

> Used React Native's StyleSheet for native style reuse on components instead of plain objects therefore I needed to change functional styles with memoize.

I didn't know about "native style reuse". I wasn't seeing any benefit in using that `Stylesheet` thing so that's why I proceeded with plain objects in the first place. Interesting, so `Stylesheet` does sound useful now, heh :) Anyway I'm ok with this.

**Comment by gwn** [comment]
ss

This is the way my Android build looks. The borders / outlines seem to be off. We agreed that Android is not our priority so an action is not really required at this point (if the iOS build is fine I mean). Just letting you know.

**Comment by gwn** [comment]
I think this is a very good looking PR in general. Thank you.

File: ios/ixoWallet.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (hunk 2 above)
Comment (Hamzali) [reply ios/ixoWallet.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:8]: After some small research, it is a new config file to track env requirement for Xcode IDE.
image
https://developer.apple.com/library/archive/releasenotes/DeveloperTools/RN-Xcode/Chapters/Introduction.html#//apple_ref/doc/uid/TP40001051-CH1-DontLinkElementID_24

File: src/lib/theme.js (hunk 6 above)
Comment (Hamzali) [reply src/lib/theme.js:5]: Yes, it will have levels of shadow something like elevation level of the surface.

File: src/assistant/Assistant.js (hunk 4 above)
Comment (gwn) [reply src/assistant/Assistant.js]: We agreed to proceed with the import syntax in our phone conversation, so resolving.

File: ios/Podfile.lock (hunk 1 above)
Comment (Hamzali) [reply ios/Podfile.lock]: reverted this change  by removing linear gradient

File: ios/ixoWallet.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (hunk 2 above)
Comment (gwn) [reply ios/ixoWallet.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:8]: wonderful thanks for the info

File: src/lib/theme.js (hunk 6 above)
Comment (gwn) [reply src/lib/theme.js:5]: ok that makes sense

**Review by gwn — CHANGES_REQUESTED** [review]
I'm okay with everything except the button styles. I can see how you based the button designs on the 3rd screen of the design (which I'm attaching to this comment for reference), but that's not ok. There's an ambiguity in the design that some buttons are bordered & inline and others are not. I guess that the styling you went with were designed for wallet addresses as a special case which I think we should ignore for now.

My request is the restyling of the buttons to make them look like the bordered & inline buttons that can be seen in the first, 4th and 5th screens of the design.

assistant

**Comment by Hamzali** [comment]
updated the button styles;
Screen Shot 2021-02-26 at 7 09 13 PM

**Comment by Hamzali** [comment]
option buttons also wrapped in a view, so it shows in one line if there is enough space for the buttons' content.
